src/test/debugger/appWorker.test.ts

```diff
@@ -6,80 +6,68 @@ import * as WebSocket from "ws";
+            const WORKER_DELAY_SHUTDOWN = `setTimeout(() => {console.log("Shutting down")}, 5000)`;
+
+            let testWorker: ForkedAppWorker;
+            let spawnStub: Sinon.SinonStub;
             let postReplyFunction = sinon.stub();
-            let readFileStub = sinon.stub();
 
-            setup(function() {
-                const nodeFileSystemMock: any = {
-                    readFile: readFileStub,
-                };
+            function workerWithScript(scriptBody: string): ForkedAppWorker {
+                const wrappedBody = [ MultipleLifetimesAppWorker.WORKER_BOOTSTRAP,
+                    scriptBody, MultipleLifetimesAppWorker.WORKER_DONE, WORKER_DELAY_SHUTDOWN ].join("\n");
 
-                const scriptImporterMock: any = {
-                    downloadAppScript: downloadAppScriptStub,
-                };
+                spawnStub = sinon.stub(child_process, "spawn", () =>
+                    originalSpawn("node", ["-e", wrappedBody], {stdio: ["pipe", "pipe", "pipe", "ipc"]}));
 
-                sandboxedWorker = new AppWorker.SandboxedAppWorker(packagerPort, sourcesStoragePath, debugAdapterPort, postReplyFunction, {
-                    nodeFileSystem: nodeFileSystemMock,
-                    scriptImporter: scriptImporterMock,
-                });
-            });
+                testWorker = new ForkedAppWorker(packagerPort, sourcesStoragePath, postReplyFunction);
+                return testWorker;
+            }
 
             teardown(function() {
                 // Reset everything
-                sandboxedWorker = null;
-
-                readFileStub = sinon.stub();
-                downloadAppScriptStub = sinon.stub();
-                postReplyFunction = sinon.stub();
+                spawnStub.restore();
+                postReplyFunction.reset();
+                if (testWorker) {
+                    testWorker.stop();
+                }
             });
 
-
             test("should execute scripts correctly and be able to invoke the callback", function() {
                 const expectedMessageResult = { success: true };
                 const startScriptContents = `var testResponse = ${JSON.stringify(expectedMessageResult)}; postMessage(testResponse);`;
 
-                readFileStub.withArgs(startScriptFileName).returns(Q.resolve(startScriptContents));
-
-                return sandboxedWorker.start().then(() => {
-                    assert(postReplyFunction.calledWithExactly(expectedMessageResult));
-                });
+                return workerWithScript(startScriptContents).start()
+                .then(() =>
+                    Q.delay(100))
+                .then(() =>
+                    assert(postReplyFunction.calledWithExactly(expectedMessageResult)));
             });
 
             test("should be able to import scripts", function() {
-                const scriptImportPath = "testScript.js";
+                // NOTE: we're not able to mock out

```

Comment:
Yeah, there should be a description of reasoning why we have to use real script file to test `importScripts`. I'll update.